Functions/Initialization/xASL_init_InitializeMutex.m

```diff
@@ -63,17 +63,99 @@
+
+    x.mutex.bAnyModuleLocked = true;
+else
+    
+    %% 4) Check if any module for this subject is locked
+    % Here we check if any other module for the same subject is locked.
+    % So whereas xASL_GoNoGo in step 2 above only checks the current
+    % module, here we check if ANY module for this subject is locked.
+
+    if ~strcmpi(x.ModuleName, 'population')
+
+        % If the current module is the population module, we don't need to
+        % perform this check because the population module is cannot be
+        % iterated over subjects, it is always run once.
+        % Plus, at the start of the population module itself, we already
+        % check if nWorkers>1, in which case we skip the population module
+        % (== the population module should not run in parallel).
+        if ~isfield(x, 'mutex') || ~isprop(x.mutex, 'Root')
+            warning('mutex field missing, something odd going wrong');
+        else
+            [~, subjectName] = fileparts(x.mutex.Root);
+            lockDir = fullfile(x.dir.xASLDerivatives, 'lock');
+						
+			% Go through all module that have lock dirs already and are thus potentially locked
+			checkModuleList = xASL_adm_GetFileList(lockDir, '.*', 'FPList', [], 1);
+            for iCheckModule=1:length(checkModuleList)
+                [~, checkModuleName] = fileparts(checkModuleList{iCheckModule});
 
+				% Verify that it is really a directory with an ASL module lock files
+                if strcmp(checkModuleName(1:12), 'xASL_module_')
+                    checkModuleName = checkModuleName(13:end); % Extract the name of the module
+                    if strcmpi(checkModuleName, x.ModuleName)
+                        % this is the current module, so we can continue if this is locked
+                    elseif strcmpi(checkModuleName, 'population') || strcmpi(checkModuleName, 'import')
+                        % for population or import modules, we can continue if other modules are locked
+                    else
+                        % For BIDS2Legacy, no visit suffix is given. For other modules, a suffix is given
+						bLockedFolders = false;
+						
+						if strcmp(x.ModuleName, 'BIDS2Legacy') && ~strcmp(checkModuleName, 'BIDS2Legacy')
+							% In case we are in BIDS2Legacy and are checking other modules, we need to add a suffix
+							
+							% List all visits
+							checkModulePath = xASL_adm_GetFileList(checkModuleList{iCheckModule}, ['^' subjectName '_\d+$'], 'FPListRec', [], 1);
+							for iVisit = 1:length(checkModulePath)
+								if ~isempty(xASL_adm_GetFileList(fullfile(checkModulePath{iVisit}, ['xASL_module_' checkModuleName]), '^locked$', 'FPListRec', [], 1))
+									bLockedFolders = true;
+								end
+							end
+						elseif ~strcmp(x.ModuleName, 'BIDS2Legacy') && strcmp(checkModuleName, 'BIDS2Legacy')
+							% In case we are in another module and checking for BIDS2Legacy, we have to remove the suffix
+							ivisitSuffix = regexp(subjectName, '_\d+$');

```

Comment:
It was correct what you had in the code - to remove suffix. But only if the current sessions wasn't BIDS2Legacy and we are checking BIDS2Legacy session. Otherwise we don't need to remove the suffix.
I have fixed that, you can resolve.